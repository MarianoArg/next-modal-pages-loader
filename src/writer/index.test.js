const mock = require('mock-fs');
const path = require('path');
const fs = require('fs');
const { writeFile } = require('./');
const { encoding } = require('../constants');

const baseDir = path.resolve(__dirname, '../../');
const outputPath = path.resolve(baseDir, 'output.js');
const moduleDir = path.resolve(baseDir);
const configFile = path.resolve(moduleDir, './module.config.json');

beforeEach(() => {
  mock({
    [outputPath]: '',
    [configFile]: '',
  });
});

afterEach(() => {
  mock.restore();
});

test('writeFile generate story loader', () => {
  const files = [
    path.resolve(__dirname, '../file1.js'),
    path.resolve(__dirname, '../sub/file2.js'),
    path.resolve(__dirname, '../../parent/file3.js'),
    path.resolve(__dirname, './sub/file4.js'),
    path.resolve(__dirname, '.\\sub\\sub\\file5.js'),
  ];
  writeFile(files, outputPath, moduleDir);

  const contents = fs.readFileSync(outputPath, encoding);

  expect(contents).toMatchSnapshot();
});

test('writeFile should generate dummy story loader if no stories were found', () => {
  const files = [];
  writeFile(files, outputPath, moduleDir);

  const contents = fs.readFileSync(outputPath, encoding);

  expect(contents).toMatchSnapshot();
});
