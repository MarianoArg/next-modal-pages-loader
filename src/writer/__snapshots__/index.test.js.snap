// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeFile generate dynamic page imports 1`] = `
"// Auto-generated file created by next-modal-pages-loader
// Do not edit.
//
import dynamic from 'next/dynamic';
import Loader from './components/loaders/loader';

const PageRoutes = new Map();

// Src/File
const SrcFile = dynamic(() => import('./src/file1'), { loading: () => <Loader /> });
PageRoutes.set(/(\\\\/.\\\\/src\\\\/file1\\\\/?$)/, <SrcFile />);

// Src/Sub/File
const SrcSubFile = dynamic(() => import('./src/sub/file2'), { loading: () => <Loader /> });
PageRoutes.set(/(\\\\/.\\\\/src\\\\/sub\\\\/file2\\\\/?$)/, <SrcSubFile />);

// Parent/File
const ParentFile = dynamic(() => import('./parent/file3'), { loading: () => <Loader /> });
PageRoutes.set(/(\\\\/.\\\\/parent\\\\/file3\\\\/?$)/, <ParentFile />);

// Writer/Sub/File
const WriterSubFile = dynamic(() => import('./src/writer/sub/file4'), { loading: () => <Loader /> });
PageRoutes.set(/(\\\\/.\\\\/src\\\\/writer\\\\/sub\\\\/file4\\\\/?$)/, <WriterSubFile />);

// Sub/Sub/File
const SubSubFile = dynamic(() => import('./src/writer/sub/sub/file5'), { loading: () => <Loader /> });
PageRoutes.set(/(\\\\/.\\\\/src\\\\/writer\\\\/sub\\\\/sub\\\\/file5\\\\/?$)/, <SubSubFile />);

export default PageRoutes;
"
`;

exports[`writeFile should generate dummy routes map if no pages were found 1`] = `
"// Auto-generated file created by next-modal-pages-loader
// Do not edit.
//
import dynamic from 'next/dynamic';
import Loader from './components/loaders/loader';

const PageRoutes = new Map();


export default PageRoutes;
"
`;
